app/Http/Controllers/MessageGroupController.php

```diff
@@ -14,6 +15,17 @@
 
 class MessageGroupController extends Controller
 {
+
+    /**
+     * Return a list of message groups that the player is in.
+     * @param Request $request
+     * @return Response
+     */
+    public function index(Request $request)
+    {
+        return new Response($this->session->player->message_groups->load('message_group_members.player'));
+    }
```

Comment:
For some reason `api/rooms/#/groups` is returning the same message groups twice. I've determined that this is because I didn't realize that when a message group is created it automatically assumes that the player sending the request should be in the group. Because of this, I have been sending the current player's id in the `participants` array, thus the user was getting added to the group twice. This made it so that when getting the list of the groups, the same group got returned multiple times because the player is in it twice.

```
[
    {
        "id": 1,
        "room_id": 4,
        "message_group_members": [
            {
                "id": 1,
                "created_at": "2020-04-25 20:49:58",
                "updated_at": "2020-04-25 20:49:58",
                "message_group_id": 1,
                "player_id": 1,
                "player": {
                    "id": 1,
                    "created_at": "2020-04-14 02:44:32",
                    "updated_at": "2020-04-25 20:40:14",
                    "name": "asdfg1",
                    "rating": 1200,
                    "wins": 0,
                    "resignations": 1,
                    "last_online_at": "2020-04-14 02:44:32"
                }
            },
            {
                "id": 2,
                "created_at": "2020-04-25 20:49:58",
                "updated_at": "2020-04-25 20:49:58",
                "message_group_id": 1,
                "player_id": 1,
                "player": {
                    "id": 1,
                    "created_at": "2020-04-14 02:44:32",
                    "updated_at": "2020-04-25 20:40:14",
                    "name": "asdfg1",
                    "rating": 1200,
                    "wins": 0,
                    "resignations": 1,
                    "last_online_at": "2020-04-14 02:44:32"
                }
            }
        ]
    },
    {
        "id": 1,
        "room_id": 4,
        "message_group_members": [
            {
                "id": 1,
                "created_at": "2020-04-25 20:49:58",
                "updated_at": "2020-04-25 20:49:58",
                "message_group_id": 1,
                "player_id": 1,
                "player": {
                    "id": 1,
                    "created_at": "2020-04-14 02:44:32",
                    "updated_at": "2020-04-25 20:40:14",
                    "name": "asdfg1",
                    "rating": 1200,
                    "wins": 0,
                    "resignations": 1,
                    "last_online_at": "2020-04-14 02:44:32"
                }
            },
            {
                "id": 2,
                "created_at": "2020-04-25 20:49:58",
                "updated_at": "2020-04-25 20:49:58",
                "message_group_id": 1,
                "player_id": 1,
                "player": {
                    "id": 1,
                    "created_at": "2020-04-14 02:44:32",
                    "updated_at": "2020-04-25 20:40:14",
                    "name": "asdfg1",
                    "rating": 1200,
                    "wins": 0,
                    "resignations": 1,
                    "last_online_at": "2020-04-14 02:44:32"
                }
            }
        ]
    },
    {
        "id": 2,
        "room_id": 4,
        "message_group_members": [
            {
                "id": 3,
                "created_at": "2020-04-26 01:26:04",
                "updated_at": "2020-04-26 01:26:04",
                "message_group_id": 2,
                "player_id": 1,
                "player": {
                    "id": 1,
                    "created_at": "2020-04-14 02:44:32",
                    "updated_at": "2020-04-25 20:40:14",
                    "name": "asdfg1",
                    "rating": 1200,
                    "wins": 0,
                    "resignations": 1,
                    "last_online_at": "2020-04-14 02:44:32"
                }
            },
            {
                "id": 4,
                "created_at": "2020-04-26 01:26:04",
                "updated_at": "2020-04-26 01:26:04",
                "message_group_id": 2,
                "player_id": 1,
                "player": {
                    "id": 1,
                    "created_at": "2020-04-14 02:44:32",
                    "updated_at": "2020-04-25 20:40:14",
                    "name": "asdfg1",
                    "rating": 1200,
                    "wins": 0,
                    "resignations": 1,
                    "last_online_at": "2020-04-14 02:44:32"
                }
            }
        ]
    },
    {
        "id": 2,
        "room_id": 4,
        "message_group_members": [
            {
                "id": 3,
                "created_at": "2020-04-26 01:26:04",
                "updated_at": "2020-04-26 01:26:04",
                "message_group_id": 2,
                "player_id": 1,
                "player": {
                    "id": 1,
                    "created_at": "2020-04-14 02:44:32",
                    "updated_at": "2020-04-25 20:40:14",
                    "name": "asdfg1",
                    "rating": 1200,
                    "wins": 0,
                    "resignations": 1,
                    "last_online_at": "2020-04-14 02:44:32"
                }
            },
            {
                "id": 4,
                "created_at": "2020-04-26 01:26:04",
                "updated_at": "2020-04-26 01:26:04",
                "message_group_id": 2,
                "player_id": 1,
                "player": {
                    "id": 1,
                    "created_at": "2020-04-14 02:44:32",
                    "updated_at": "2020-04-25 20:40:14",
                    "name": "asdfg1",
                    "rating": 1200,
                    "wins": 0,
                    "resignations": 1,
                    "last_online_at": "2020-04-14 02:44:32"
                }
            }
        ]
    }
]
```

Should add a check on ids sent in from the participant array to prevent adding yourself.

Reply:
Converted to issue #26. Resolving